pkg/utl/rediscache/redis_test.go

```diff
@@ -96,6 +106,14 @@ func TestSetKeyValue(t *testing.T) {
 					return nil, fmt.Errorf("some error")
 				})
 			}
+
+			if tt.jsonErr {
+				patchJson := ApplyFunc(json.Marshal, func(v any) ([]byte, error) {
+					return []byte{}, fmt.Errorf("json error")
```

Comment:
at the response add a check to see if the json error is being thrown at the response error when tt.jsonErr is true